File: 121.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 0) return 0;
        int cur = prices[0], ans = 0;
        for(int i = 1;i < prices.length;i ++) {
            ans = Math.max(ans, prices[i] - cur);
            cur = Math.min(cur, prices[i]);
        }
        return ans;
    }
}

File: 121.go
func maxProfit(prices []int) int {
	min_price := prices[0]
	max_profit := 0
	for _, v := range prices {
		max_profit = max(max_profit, v-min_price)
		min_price = min(min_price, v)
	}
	return max_profit
}

func max(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}

func min(a, b int) int {
	if a < b {
		return a
	} else {
		return b
	}
}